src/runner.jl

```diff
@@ -167,28 +175,66 @@ function test!(pkg::AbstractString,
     if !isfile(testfilepath)
         push!(notests, pkg)
     else
-        @info "Testing $pkg"
         logfilename = joinpath(logfilepath, "testlog.xml") # TODO handle having multiple packages called on after the other
-        runner_file_path = make_runner_file(testfilepath, logfilename)
-        cd(dirname(testfilepath)) do
-            try
-                cmd = ```
-                    $(Base.julia_cmd())
-                    --code-coverage=$(coverage ? "user" : "none")
-                    --color=$(Base.have_color ? "yes" : "no")
-                    --compiled-modules=$(Bool(Base.JLOptions().use_compiled_modules) ? "yes" : "no")
-                    --check-bounds=yes
-                    --depwarn=$(Base.JLOptions().depwarn == 2 ? "error" : "yes")
-                    --inline=$(Bool(Base.JLOptions().can_inline) ? "yes" : "no")
-                    --startup-file=$(Base.JLOptions().startupfile != 2 ? "yes" : "no")
-                    --track-allocation=$(("none", "user", "all")[Base.JLOptions().malloc_log + 1])
-                    $(runner_file_path)
-                    ```
-                run(cmd)
-                @info "$pkg tests passed. Results saved to $logfilename."
-            catch err
-                @warn "ERROR: Test(s) failed or had an error in $pkg"
-                push!(errs,pkg)
+        testreportsdir = dirname(@__DIR__)
+        runner_code = gen_runner_code(testfilepath, logfilename, testreportsdir)
+        cmd = ```
+            $(Base.julia_cmd())
+            --code-coverage=$(coverage ? "user" : "none")
+            --color=$(Base.have_color ? "yes" : "no")
+            --compiled-modules=$(Bool(Base.JLOptions().use_compiled_modules) ? "yes" : "no")
+            --check-bounds=yes
+            --depwarn=$(Base.JLOptions().depwarn == 2 ? "error" : "yes")
+            --inline=$(Bool(Base.JLOptions().can_inline) ? "yes" : "no")
+            --startup-file=$(Base.JLOptions().startupfile != 2 ? "yes" : "no")
+            --track-allocation=$(("none", "user", "all")[Base.JLOptions().malloc_log + 1])
+            --eval $(runner_code)
+            ```
+
+        test_folder_has_project_file = isfile(replace(testfilepath, "runtests.jl"=>"Project.toml"))
```

Comment:
Maybe tp be sure it is not a very weird path or package name:
```suggestion
        test_folder_has_project_file = isfile(replace(testfilepath, r"/runtests.jl$"=>"/Project.toml"))
```